subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -30,18 +31,26 @@ class DeviceWorker(
         Executors.newSingleThreadExecutor().asCoroutineDispatcher()
     )
 
-    private var state: State = device.state()
-
     fun run() = scope.launch {
 
+        var state: State = try {
```

Comment:
Could we replace `var` with two `val`
`previousState` and `preparedState`

Reply:
Can't get rid of mutable state. It's mutated in the loop on each cycle.